test/test.dart

```diff
@@ -710,3 +710,16 @@ Future<void> enableAllAutomaticRecovery() async {
     await client.setAutomaticRecoveryEnabled(appName, true);
   }
 }
+
+
+extension $WriteExtension on Realm {
+  void writesExpectException<TException>(String exceptionMessage) {
```

Comment:
Unlike .NET we are not really naming type arguments like that `TException`. I would suggest doing something like 
```suggestion
  void writesExpectException<T extends Exception>(String message) {
```

Reply:
We have one case that throws `RealmClosedError`. It is not an Exception. This is in test  'Realm write after realm is closed'. I will just rename `TException` to `T`. Or maybe `RealmClosedError` should have been Exception?

Reply:
This test should not use `writesExpectException`. Lets this method only in the tests that require the double writes.

Reply:
I'm leaving this with <T> because of `RealmClosedError`